### PR TITLE
REF: share methods between ExtensionBlock, NDArrayBackedExtensionBlock

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@jreback adds test discussed in #40763

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -1354,6 +1354,31 @@ def delete(self, loc) -> None:
             # _cache not yet initialized
             pass
 
+    @cache_readonly
+    def array_values(self) -> ExtensionArray:
+        return self.values
+
+    def get_values(self, dtype: DtypeObj | None = None) -> np.ndarray:
```
Comment (jreback) [inline pandas/core/internals/blocks.py:1361]: do we not use Optional anymore?

**Review by jreback — APPROVED** [review]
great (minor question)

File: pandas/core/internals/blocks.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:1361]: i think a code check was added in the last few days, not sure of the specifics

**Pull request merged**